### ISSUE TITLE
Inspection for wrap array_filter with array_values to not lose numerical keys

### ISSUE THREAD

**Issue by Haehnchen** [kalessil/phpinspectionsea#631]
```php
$x = ['foobar', 'foo'];

$var = array_filter($x, function($y) {
    return $y == 'foo';
});

# now get first result:
echo $var[0]; // but its $var[1]

```

So in first place an inspection for something like this maybe helpful

```php
array_values(array_filter($x, function($y) {
    return $y == 'foo';
}));
```

**Comment by rentalhost** [comment]
I don't know if it should be an inspection. There are cases where it is expected, so how EA could differentiate this?

**Comment by kalessil** [comment]
Obviously, we have an interesting case here - I'll do research on a couple of major frameworks to see how often we are dealing with this situation.

**Comment by kalessil** [comment]
@Haehnchen : was the original case also using hardcoded indexes when accessing the filtered result?

**Comment by funivan** [comment]
Maybe inspection can be applied when we have filter and then access by key.

**Comment by kalessil** [comment]
In general yes, the question is if numeric indexes hardcoded or not.

With string indexes, the case is not a problem, as speaking itself associative array keys are preserved.
But integer indexes can lead to error, I'm trying to find the original case.

@Haehnchen do you have an original case available for research?

**Comment by kalessil** [comment]
Criteria:
- after `$variable = array_filter(...)`
- following `$variable[<index expression>]`
